References **[Pandas Home Page](https://www.kaggle.com/learn/pandas)**

---


Run the following cell to load the bangalore restaurants data 

In [2]:
import pandas as pd
import numpy as np

restaurants = pd.read_csv('data/zomato.csv')

Look at an overview of your data by running the following line.


In [2]:
restaurants.head()

url  \
0  https://www.zomato.com/bangalore/jalsa-banasha...   
1  https://www.zomato.com/bangalore/spice-elephan...   
2  https://www.zomato.com/SanchurroBangalore?cont...   
3  https://www.zomato.com/bangalore/addhuri-udupi...   
4  https://www.zomato.com/bangalore/grand-village...   

                                             address                   name  \
0  942, 21st Main Road, 2nd Stage, Banashankari, ...                  Jalsa   
1  2nd Floor, 80 Feet Road, Near Big Bazaar, 6th ...         Spice Elephant   
2  1112, Next to KIMS Medical College, 17th Cross...        San Churro Cafe   
3  1st Floor, Annakuteera, 3rd Stage, Banashankar...  Addhuri Udupi Bhojana   
4  10, 3rd Floor, Lakshmi Associates, Gandhi Baza...          Grand Village   

  online_order book_table   rate  votes                             phone  \
0          Yes        Yes  4.1/5    775    080 42297555\r\n+91 9743772233   
1          Yes         No  4.1/5    787                      080 41714161   
2          Yes         No  3.8/5    918                    +91 9663487993   
3           No         No  3.7/5     88                    +91 9620009302   
4           No         No  3.8/5    166  +91 8026612447\r\n+91 9901210005   

       location            rest_type  \
0  Banashankari        Casual Dining   
1  Banashankari        Casual Dining   
2  Banashankari  Cafe, Casual Dining   
3  Banashankari          Quick Bites   
4  Basavanagudi        Casual Dining   

                                          dish_liked  \
0  Pasta, Lunch Buffet, Masala Papad, Paneer Laja...   
1  Momos, Lunch Buffet, Chocolate Nirvana, Thai G...   
2  Churros, Cannelloni, Minestrone Soup, Hot Choc...   
3                                        Masala Dosa   
4                                Panipuri, Gol Gappe   

                         cuisines approx_cost(for two people)  \
0  North Indian, Mughlai, Chinese                         800   
1     Chinese, North Indian, Thai                         800   
2          Cafe, Mexican, Italian                         800   
3      South Indian, North Indian                         300   
4        North Indian, Rajasthani                         600   

                                        reviews_list menu_item  \
0  [('Rated 4.0', 'RATED\n  A beautiful place to ...        []   
1  [('Rated 4.0', 'RATED\n  Had been here for din...        []   
2  [('Rated 3.0', "RATED\n  Ambience is not that ...        []   
3  [('Rated 4.0', "RATED\n  Great food and proper...        []   
4  [('Rated 4.0', 'RATED\n  Very good restaurant ...        []   

  listed_in(type) listed_in(city)  
0          Buffet    Banashankari  
1          Buffet    Banashankari  
2          Buffet    Banashankari  
3          Buffet    Banashankari  
4          Buffet    Banashankari

# Exercises

## 1.

Select the `online_order` column from `restaurants` and assign the result to the variable `ol_order`.

In [16]:
# Your code here
ol_order = restaurants["online_order"]

# Check your answer
ol_order

0        Yes
1        Yes
2        Yes
3         No
4         No
        ... 
51712     No
51713     No
51714     No
51715     No
51716     No
Name: online_order, Length: 51717, dtype: object

What type of object is `ol_order`? If you're not sure, you can check by calling Python's `type` function: `type(ol_order)`.

In [17]:
type(ol_order)

pandas.core.series.Series

## 2.

Select the first value from the `dish_liked` column of `restaurants`, assigning it to variable `first_rest_reviews`.

<br><br> Output should be:<br>`'Pasta, Lunch Buffet, Masala Papad, Paneer Lajawab, Tomato Shorba, Dum Biryani, Sweet Corn Soup'`

In [4]:
first_rest_reviews = restaurants.dish_liked[0]

# Check your answer

first_rest_reviews

'Pasta, Lunch Buffet, Masala Papad, Paneer Lajawab, Tomato Shorba, Dum Biryani, Sweet Corn Soup'

## 3. 

Select the first row of data (the first record) from `restaurants`, assigning it to the variable `first_row`.

In [5]:
first_row = restaurants.iloc[0, :]


first_row

url                            https://www.zomato.com/bangalore/jalsa-banasha...
address                        942, 21st Main Road, 2nd Stage, Banashankari, ...
name                                                                       Jalsa
online_order                                                                 Yes
book_table                                                                   Yes
rate                                                                       4.1/5
votes                                                                        775
phone                                             080 42297555\r\n+91 9743772233
location                                                            Banashankari
rest_type                                                          Casual Dining
dish_liked                     Pasta, Lunch Buffet, Masala Papad, Paneer Laja...
cuisines                                          North Indian, Mughlai, Chinese
approx_cost(for two people) 

## 4.

Select the first 10 values from the `approx_cost(for two people)` column in `restaurants`, assigning the result to variable `cost_selection`.

Hint: format your output as a pandas Series.

In [28]:
#cost_selection = restaurants["approx_cost(for two people)"].iloc[:10]
cost_selection = restaurants.loc[:9, "approx_cost(for two people)"]

# Check your answer

cost_selection

0    800
1    800
2    800
3    300
4    600
5    600
6    800
7    600
8    700
9    550
Name: approx_cost(for two people), dtype: object

## 5.

Select the records with index labels `2`, `4`, `6`, `8`, and `10`, assigning the result to the variable `sub_set`.

Should look like this:
<div>
<img src="images/subset.png" width="750"/>
</div>

In [6]:
sub_set = restaurants.loc[[2, 4, 6, 8, 10]]

# Check your answer
sub_set

url  \
2   https://www.zomato.com/SanchurroBangalore?cont...   
4   https://www.zomato.com/bangalore/grand-village...   
6   https://www.zomato.com/bangalore/rosewood-inte...   
8   https://www.zomato.com/bangalore/penthouse-caf...   
10  https://www.zomato.com/bangalore/caf%C3%A9-dow...   

                                              address  \
2   1112, Next to KIMS Medical College, 17th Cross...   
4   10, 3rd Floor, Lakshmi Associates, Gandhi Baza...   
6   19/1, New Timberyard Layout, Beside Satellite ...   
8   1, 30th Main Road, 3rd Stage, Banashankari, Ba...   
10  12,29 Near PES University Back Gate, D'Souza N...   

                                                 name online_order book_table  \
2                                     San Churro Cafe          Yes         No   
4                                       Grand Village           No         No   
6     Rosewood International Hotel - Bar & Restaurant           No         No   
8                                      Penthouse Cafe          Yes         No   
10  Café Down The A...          Yes         No   

     rate  votes                             phone      location  \
2   3.8/5    918                    +91 9663487993  Banashankari   
4   3.8/5    166  +91 8026612447\r\n+91 9901210005  Basavanagudi   
6   3.6/5      8    +91 9731716688\r\n080 26740366   Mysore Road   
8   4.0/5    324  +91 8884135549\r\n+91 9449449316  Banashankari   
10  4.1/5    402    080 26724489\r\n+91 7406048982  Banashankari   

              rest_type                                         dish_liked  \
2   Cafe, Casual Dining  Churros, Cannelloni, Minestrone Soup, Hot Choc...   
4         Casual Dining                                Panipuri, Gol Gappe   
6         Casual Dining                                                NaN   
8                  Cafe  Pizza, Mocktails, Coffee, Nachos, Salad, Pasta...   
10                 Cafe  Waffles, Pasta, Crispy Chicken, Honey Chilli C...   

                                       cuisines approx_cost(for two people)  \
2                        Cafe, Mexican, Italian                         800   
4                      North Indian, Rajasthani                         600   
6   North Indian, South Indian, Andhra, Chinese                         800   
8                    Cafe, Italian, Continental                         700   
10                                         Cafe                         500   

                                         reviews_list menu_item  \
2   [('Rated 3.0', "RATED\n  Ambience is not that ...        []   
4   [('Rated 4.0', 'RATED\n  Very good restaurant ...        []   
6   [('Rated 5.0', 'RATED\n  Awesome food ??Great ...        []   
8   [('Rated 3.0', "RATED\n  I had been to this pl...        []   
10  [('Rated 4.0', 'RATED\n  We ended up here on a...        []   

   listed_in(type) listed_in(city)  
2           Buffet    Banashankari  
4           Buffet    Banashankari  
6           Buffet    Banashankari  
8            Cafes    Banashankari  
10           Cafes    Banashankari

Use the len() function to determine how many reviews our csv contains - (51717)

In [31]:
len(restaurants)

51717

If we want to sample our data we may decide to take every nth data point.<br><br> Select every 1000th entry in our dataframe.<br> Do this by using an array of the values - `[0,1000,2000,3000,......49000,50000,51000]` <br>(Hint - You should be able to do this using np.arange - this will require importing the numpy library!)

In [36]:
fixed_selection = restaurants.iloc[::1000, :]
# fixed_selection = np.arange(0,len(restaurants),1000)
fixed_selection



url  \
0      https://www.zomato.com/bangalore/jalsa-banasha...   
1000   https://www.zomato.com/bangalore/andhra-ruchul...   
2000   https://www.zomato.com/bangalore/guru-garden-b...   
3000   https://www.zomato.com/bangalore/kudla-hotel-r...   
4000   https://www.zomato.com/bangalore/uptown-slider...   
5000   https://www.zomato.com/bangalore/red-onion-sha...   
6000   https://www.zomato.com/bangalore/dunkin-donuts...   
7000   https://www.zomato.com/bangalore/curries-pickl...   
8000   https://www.zomato.com/bangalore/slounge-lemon...   
9000   https://www.zomato.com/bangalore/punjabi-corne...   
10000  https://www.zomato.com/bangalore/north-indian-...   
11000  https://www.zomato.com/bangalore/crafted-plate...   
12000  https://www.zomato.com/bangalore/backyard-rich...   
13000  https://www.zomato.com/bangalore/chaitanya-veg...   
14000  https://www.zomato.com/bangalore/cake-menu-1-e...   
15000  https://www.zomato.com/bangalore/happy-filling...   
16000  https://www.zomato.com/bangalore/la-pinoz-pizz...   
17000  https://www.zomato.com/bangalore/third-wave-co...   
18000  https://www.zomato.com/bangalore/petoo-old-mad...   
19000  https://www.zomato.com/bangalore/dcafe-domlur-...   
20000  https://www.zomato.com/bangalore/lassi-shop-jp...   
21000  https://www.zomato.com/bangalore/rock-on-rolls...   
22000  https://www.zomato.com/bangalore/jaganath-hote...   
23000  https://www.zomato.com/bangalore/multi-cakes-j...   
24000  https://www.zomato.com/bangalore/janatha-cafe-...   
25000  https://www.zomato.com/bangalore/sankys-burger...   
26000  https://www.zomato.com/bangalore/vanas-kammana...   
27000  https://www.zomato.com/bangalore/pasta-street-...   
28000  https://www.zomato.com/bangalore/vegan-khana-k...   
29000  https://www.zomato.com/bangalore/avon-bar-kitc...   
30000  https://www.zomato.com/bangalore/anikas-india-...   
31000  https://www.zomato.com/bangalore/cream-n-desse...   
32000  https://www.zomato.com/bangalore/the-fixx-kora...   
33000  https://www.zomato.com/bangalore/cafe-221b-btm...   
34000  https://www.zomato.com/bangalore/banjara-melti...   
35000  https://www.zomato.com/bangalore/alankrutha-bt...   
36000  https://www.zomato.com/bangalore/truffles-kora...   
37000  https://www.zomato.com/bangalore/pizza-stop-ko...   
38000  https://www.zomato.com/bangalore/the-wok-shop-...   
39000  https://www.zomato.com/bangalore/ram-prasad-ve...   
40000  https://www.zomato.com/bangalore/juglee-new-be...   
41000  https://www.zomato.com/bangalore/longlive-pizz...   
42000  https://www.zomato.com/bangalore/chaat-street-...   
43000  https://www.zomato.com/bangalore/hyper-scoop-s...   
44000  https://www.zomato.com/bangalore/h-w-cafe-new-...   
45000  https://www.zomato.com/bangalore/bombay-brasse...   
46000  https://www.zomato.com/bangalore/waah-chef-1-c...   
47000  https://www.zomato.com/bangalore/bhairaveshwar...   
48000  https://www.zomato.com/bangalore/koshys-richmo...   
49000  https://www.zomato.com/bangalore/udupi-savi-th...   
50000  https://www.zomato.com/bangalore/costa-coffee-...   
51000  https://www.zomato.com/bangalore/yummtumm-whit...   

                                                 address  \
0      942, 21st Main Road, 2nd Stage, Banashankari, ...   
1000   1319, 100 Feet Road, 2nd Phase, JP Nagar, Bang...   
2000   197, Next to Reliance Mart, Opposite Accenture...   
3000   Hotel Ramanashree, 16, Raja Ram Mohan Roy Road...   
4000   647, Ground Floor, 27th Main Road, Sector 1, H...   
5000   Money Chambers Double Road, Shanti Nagar, Bang...   
6000          Mota Royal Arcade, Brigade Road, Bangalore   
7000   1376, 6th Main, E Block, AECS Layout, Brookefi...   
8000   Lemon Tree Hotel, 23, EPIP Zone, Near I Gate C...   
9000   13/11, 1A, BVN Building, Arekere Main Road, Ba...   
10000  188, 5th Cross, KHB Colony, Koramangala 5th Bl...   
11000  1304, 1st Floor, 9th Cross, 20th Main, 2nd Pha...   
12000  19, Anjaneya Temple Street, Y G Palya, Austin ...   
13000  M.F Noranha Road, Opp. Ru

It would be more correct to take an absolutely random selection - as there may be an ordering in the data we are not aware of.<br><br> Select every 50 completely random entries<br> Do this by using an array of random values between 0 and 51717 <br>(Hint - You should be able to do this using the numpy library you already imported!)

In [7]:
random_selection = np.random.randint(51517, size=50)

r = restaurants.loc[random_selection]
r


url  \
5209   https://www.zomato.com/bangalore/the-waffle-pr...   
44038  https://www.zomato.com/bangalore/zam-zam-kabab...   
18247  https://www.zomato.com/bangalore/huga-icecream...   
48136  https://www.zomato.com/bangalore/taste-of-kera...   
25664  https://www.zomato.com/bangalore/manjit-da-dha...   
36748  https://www.zomato.com/bangalore/pot-o-noodles...   
50025  https://www.zomato.com/bangalore/inhouse-burge...   
50249  https://www.zomato.com/bangalore/andhra-kafe-w...   
37658  https://www.zomato.com/bangalore/skywalk-resto...   
38666  https://www.zomato.com/bangalore/bangalore-pub...   
25264  https://www.zomato.com/bangalore/srinidhi-veg-...   
22829  https://www.zomato.com/bangalore/fritz-haber-b...   
33343  https://www.zomato.com/bangalore/stoked-korama...   
27257  https://www.zomato.com/bangalore/chillout-btm-...   
42060  https://www.zomato.com/bangalore/rekha-restaur...   
5326   https://www.zomato.com/bangalore/double-decker...   
22790  https://www.zomato.com/bangalore/odhavji-kitch...   
25764  https://www.zomato.com/bangalore/kabsa-house-k...   
39106  https://www.zomato.com/bangalore/chinese-corne...   
12757  https://www.zomato.com/bangalore/saatvikk-brig...   
37449  https://www.zomato.com/bangalore/martinos-kora...   
5232   https://www.zomato.com/bangalore/wafl-caf%C3%A...   
2418   https://www.zomato.com/bangalore/the-hungers-z...   
40077  https://www.zomato.com/bangalore/sri-venkatesh...   
43623  https://www.zomato.com/bangalore/radhika-veg-d...   
43275  https://www.zomato.com/bangalore/alchemy-the-c...   
14212  https://www.zomato.com/bangalore/andhra-mess-e...   
15794  https://www.zomato.com/bangalore/barbeque-nati...   
48071  https://www.zomato.com/bangalore/santhanam-swe...   
7447   https://www.zomato.com/bangalore/indulge-by-in...   
44118  https://www.zomato.com/bangalore/saha-chowmein...   
22618  https://www.zomato.com/bangalore/mane-biryani-...   
30926  https://www.zomato.com/bangalore/lot-like-crep...   
30356  https://www.zomato.com/bangalore/ober-cafe-kor...   
28275  https://www.zomato.com/bangalore/mr-meetharam-...   
10958  https://www.zomato.com/bangalore/tunday-kababi...   
25437  https://www.zomato.com/bangalore/paaramparya-r...   
16347  https://www.zomato.com/bangalore/mamas-cuisine...   
16024  https://www.zomato.com/bangalore/anjappar-hsr?...   
20172  https://www.zomato.com/bangalore/mecaf-family-...   
45917  https://www.zomato.com/bangalore/chef-inn-rege...   
18101  https://www.zomato.com/bangalore/desi-streat-b...   
39097  https://www.zomato.com/bangalore/shiv-sager-br...   
45403  https://www.zomato.com/bangalore/sree-vinayaka...   
49763  https://www.zomato.com/bangalore/polar-bear-sa...   
51502  https://www.zomato.com/bangalore/nys-kitchen-w...   
28036  https://www.zomato.com/bangalore/oveanly-btm?c...   
37262  https://www.zomato.com/bangalore/veruthe-oru-t...   
23842  https://www.zomato.com/bangalore/turquoise-jp-...   
38844  https://www.zomato.com/bangalore/hashtag-cafe-...   

                                                 address  \
5209   Shop 3 & 14, Shanthala Nagar, Ashok Nagar, St....   
44038  40/7, Sri Sai Complex, MSRCE College Road, Mat...   
18247  184, 1st Floor, Opposite Kaveri School, Indira...   
48136           52, 1st Floor, Residency Road, Bangalore   
25664  107, 1st Floor, Wheeler Road, Near Railway Cro...   
36748  618, 80 Feet Road, Koramangala 4th Block, Bang...   
50025  Inside Pritech Park, shop no 5,Near RMZ Eco Sp...   
50249  Opposite Prestige Shantiniketan, Hoodi Village...   
37658  169, 80 Feet Road, Koramangala 1st Block, Bang...   
38666  2nd Floor, 1 MG Mall, Trinity Circle, MG Road,...   
25264  7th Main, 80 Feet Road, 1st Block, HRBR Layout...   
22829                                      Delivery Only   
33343    458, 8th Main, Koramangala 4th Block, Bangalore   
27257  8, G Cross, 20th Main Road, BTM 1st stage, BTM...   
42060  39, Babbu Reddy Building, Kundanahalli Gate, B...   
5326   120, 4th Floor, Lancer Bu

## 6.

Create a variable `dataframe` containing the `address`, `name`, `book_table`,`rate`,and `listed_in(city)` columns of the records with the index labels `0`, `5`, `50`,`500`, and `5000`. In other words, generate the following DataFrame:

![](images/subset2.png)

In [52]:
dataframe = restaurants.loc[[0, 5, 50, 500, 5000, 50000], ["address", "name", "book_table", "rate", "listed_in(city)"]]

# Check your answer
dataframe

address             name  \
0      942, 21st Main Road, 2nd Stage, Banashankari, ...            Jalsa   
5      37, 5-1, 4th Floor, Bosco Court, Gandhi Bazaar...  Timepass Dinner   
50     276, Ground Floor, 100 Feet Outer Ring Road, B...            Petoo   
500    374/B, 2nd Phase, BSK 3rd Stage, Srinivasa Nag...     Kanti Sweets   
5000   Money Chambers Double Road, Shanti Nagar, Bang...        Red Onion   
50000  RMZ Eco Space, Varthur Hobli, Bellandur, Banga...     Costa Coffee   

      book_table    rate listed_in(city)  
0            Yes   4.1/5    Banashankari  
5             No   3.8/5    Banashankari  
50            No   3.7/5    Banashankari  
500           No   4.0/5    Banashankari  
5000         Yes  4.2 /5    Brigade Road  
50000         No  3.0 /5   Sarjapur Road

## 7.

Create a variable `df` containing the `address`, `name`, `book_table`,`rate`, and `listed_in(city)` columns of the first 100 records. 

Hint: you may use `loc` or `iloc`. When working on the answer this question and the several of the ones that follow, keep the following "gotcha" described in the tutorial:

> `iloc` uses the Python stdlib indexing scheme, where the first element of the range is included and the last one excluded. 
`loc`, meanwhile, indexes inclusively. 

> This is particularly confusing when the DataFrame index is a simple numerical list, e.g. `0,...,1000`. In this case `df.iloc[0:1000]` will return 1000 entries, while `df.loc[0:1000]` return 1001 of them! To get 1000 elements using `loc`, you will need to go one lower and ask for `df.iloc[0:999]`. 

In [53]:
df = restaurants.loc[:100, ["address", "name", "book_table", "rate", "listed_in(city)"]]
# Check your answer
df

address                   name  \
0    942, 21st Main Road, 2nd Stage, Banashankari, ...                  Jalsa   
1    2nd Floor, 80 Feet Road, Near Big Bazaar, 6th ...         Spice Elephant   
2    1112, Next to KIMS Medical College, 17th Cross...        San Churro Cafe   
3    1st Floor, Annakuteera, 3rd Stage, Banashankar...  Addhuri Udupi Bhojana   
4    10, 3rd Floor, Lakshmi Associates, Gandhi Baza...          Grand Village   
..                                                 ...                    ...   
96   50 Feet Road, Hanumanthanagar, Banashankari, B...                 Kaggis   
97   34, 22nd Main, 22nd Cross, Opposite BDA, 2nd S...   Ayda Persian Kitchen   
98   24/1, 23rd Cross, 6th Main Road, Opposite Kave...           Chatar Patar   
99   96, Opposite Jain Collage, Basavanagudi, Banga...             Polar Bear   
100  2465, 3rd Floor, Opposite BDA Complex, 24th Cr...      Kidambi's Kitchen   

    book_table   rate listed_in(city)  
0          Yes  4.1/5    Banashankari  
1           No  4.1/5    Banashankari  
2           No  3.8/5    Banashankari  
3           No  3.7/5    Banashankari  
4           No  3.8/5    Banashankari  
..         ...    ...             ...  
96          No  3.8/5    Banashankari  
97          No  3.7/5    Banashankari  
98          No  3.7/5    Banashankari  
99          No  3.8/5    Banashankari  
100         No  3.5/5    Banashankari  

[101 rows x 5 columns]

## 8.

Create a DataFrame `restaurants` in `Bellandur`. Hint: `restaurants.location` equals what?

In [57]:
bellandur_restaurants = restaurants.loc[restaurants.location=="Bellandur"]

# bellandur_restaurants = restaurants[restaurants.location=="Bellandur"]

# Check your answer
bellandur_restaurants

url  \
3746   https://www.zomato.com/bangalore/kava-fairfiel...   
3748   https://www.zomato.com/bangalore/momo-cafe-cou...   
3749   https://www.zomato.com/bangalore/ruh-bellandur...   
3750   https://www.zomato.com/bangalore/tipsy-bull-th...   
3751   https://www.zomato.com/bangalore/nook-aloft-be...   
...                                                  ...   
50096  https://www.zomato.com/bangalore/atithi-family...   
50134  https://www.zomato.com/bangalore/re-fuel-aloft...   
51033  https://www.zomato.com/bangalore/re-fuel-aloft...   
51628  https://www.zomato.com/bangalore/nook-aloft-be...   
51633  https://www.zomato.com/bangalore/re-fuel-aloft...   

                                                 address  \
3746   Fairfield by Marriott, Marathahalli - Sarjapur...   
3748   Courtyard by Marriott, Marathahalli-Sarjapur O...   
3749   4th Floor, Ganapa Tower, Sarjapur Outer Ring R...   
3750     24-26, 414/80/2, Varthur Hobli, Outer Ring Road   
3751   Cessna Business Park, Sarjapur ? Marathahalli ...   
...                                                  ...   
50096  289, Next to Central Mall, Outer Ring Road, Be...   
50134  Sarjapura - Marathahalli Outer Ring Road, Kadu...   
51033  Sarjapura - Marathahalli Outer Ring Road, Kadu...   
51628  Cessna Business Park, Sarjapur ÃÂÂÂ...   
51633  Sarjapura - Marathahalli Outer Ring Road, Kadu...   

                                                name online_order book_table  \
3746                    Kava - Fairfield by Marriott           No        Yes   
3748               MoMo Cafe - Courtyard by Marriott           No        Yes   
3749                                             Ruh          Yes        Yes   
3750                   Tipsy Bull - The Bar Exchange          Yes        Yes   
3751     Nook - Aloft Bengaluru Cessna Business Park           No        Yes   
...                                              ...          ...        ...   
50096                 Atithi Family Bar & Restaurant          Yes         No   
50134  Re:fuel- Aloft Bengaluru Cessna Business Park           No         No   
51033  Re:fuel- Aloft Bengaluru Cessna Business Park           No         No   
51628    Nook - Aloft Bengaluru Cessna Business Park           No        Yes   
51633  Re:fuel- Aloft Bengaluru Cessna Business Park           No         No   

         rate  votes                             phone   location  \
3746    3.9/5    284  +91 9513584683\r\n+91 9538877533  Bellandur   
3748    4.2/5    271  +91 9513653156\r\n+91 9538877533  Bellandur   
3749    4.1/5   3020  +91 9972305871\r\n+91 9972934777  Bellandur   
3750    4.4/5   2182                      080 49653133  Bellandur   
3751    4.3/5    403  +91 9663517066\r\n+91 9686861135  Bellandur   
...       ...    ...                               ...        ...   
50096  3.6 /5     31    +91 9008008060\n+91 9902704050  Bellandur   
50134     NaN      0                      080 45101033  Bellandur   
51033     NaN      0                      080 45101033  Bellandur   
51628  4.2 /5    411    +91 9663517066\n+91 9686861135  Bellandur   
51633     NaN      0                      080 45101033  Bellandur   

                rest_type                                         dish_liked  \
3746        Casual Dining  Tea, Salads, Fish, Chicken Biryani, Breakfast ...   
3748        Casual Dining  Pasta, Lunch Buffet, Butter Chicken, Salads, B...   
3749        Casual Dining  Fattoush Salad, Mutton Biryani, Mutton Roganjo...   
3750   Bar, Casual Dining  Falafel, Cocktails, Mocktails, Sea Food, Sprin...   
3751        Casual Dining  Chicken Quesadilla, Breakfast Buffet, Cheeseca...   
...                   ...                                                ...   
50096  Casual Dining, Bar                                                NaN   
50134        Cafe, Bakery                                                NaN   
51033        Cafe, Bakery                                                NaN   
51628       Casual Di

## 9.

Create a DataFrame `Electronic_City_takeaways` containing all restaurants in Electronic City that take online orders

In [65]:
Electronic_City_takeaways = restaurants.loc[restaurants.online_order=="Yes"].loc[restaurants.location=="Electronic City"]

# Electronic_City_takeaways = restaurants.loc[(restaurants.location=="Electronic City") & (restaurants.online_order=="Yes")]

# Check your answer
Electronic_City_takeaways

url  \
1684   https://www.zomato.com/bangalore/ministry-of-b...   
2395   https://www.zomato.com/bangalore/zatar-house-1...   
13370  https://www.zomato.com/bangalore/onesta-electr...   
13371  https://www.zomato.com/bangalore/time-travelle...   
13376  https://www.zomato.com/bangalore/aagaman-veg-e...   
...                                                  ...   
17397  https://www.zomato.com/bangalore/nmr-multicuis...   
17409  https://www.zomato.com/bangalore/godavari-deli...   
17475  https://www.zomato.com/bangalore/halli-mane-ba...   
17479  https://www.zomato.com/bangalore/biryani-darba...   
23602  https://www.zomato.com/bangalore/zatar-house-1...   

                                                 address  \
1684   No 121/1, Abhinandana Complex, Manipal Country...   
2395   V-8, 3rd Cross, Keonics Road, Electronic City,...   
13370  Sy No: 54/2 ( Old Sy no 54 ), Above BATA showr...   
13371  Sai Vishram Business Hotel, 144 - C/1, Kirlosk...   
13376  Legacy Building, Ground Floor, Behind Niligiri...   
...                                                  ...   
17397  200/1, Kudlu Main Road, Near Corporation Bank,...   
17409  44/45, Behind Hotel Ekaa, Krishna Reddy Indust...   
17475  AECS Layout - A Block, Singasandra, Electronic...   
17479  44/45 Behind Hotel Ekaa, Krishna Reddy Industr...   
23602  V-8, 3rd Cross, Keonics Road, Electronic City,...   

                               name online_order book_table   rate  votes  \
1684             Ministry of Burger          Yes         No    NaN      0   
2395                    Zatar House          Yes         No  4.0/5     25   
13370                        Onesta          Yes        Yes  4.5/5    860   
13371  Time Traveller - Sai Vishram          Yes         No  4.2/5    882   
13376                   Aagaman Veg          Yes         No    NEW      0   
...                             ...          ...        ...    ...    ...   
17397   NMR Multicuisine Restaurant          Yes         No  3.6/5      9   
17409             Godavari Delights          Yes         No  3.0/5     11   
17475      Halli Mane Badoota Hotel          Yes         No    NEW      0   
17479                Biryani Darbar          Yes         No  3.3/5     12   
23602                   Zatar House          Yes         No  4.0/5     32   

                                  phone         location            rest_type  \
1684   +91 8750482000\r\n+91 9910009994  Electronic City             Delivery   
2395   +91 7338538555\r\n+91 7338526555  Electronic City        Casual Dining   
13370                      080 41177515  Electronic City  Casual Dining, Cafe   
13371      080 40400400\r\n00 997200400  Electronic City        Casual Dining   
13376  +91 9632145035\r\n+91 9606303905  Electronic City        Casual Dining   
...                                 ...              ...                  ...   
17397                    +91 9686769987  Electronic City        Casual Dining   
17409                    +91 8884899955  Electronic City        Casual Dining   
17475                    +91 9900744265  Electronic City          Quick Bites   
17479                    +91 8884299936  Electronic City        Casual Dining   
23602  +91 7338538555\r\n+91 7338526555  Electronic City        Casual Dining   

                                              dish_liked  \
1684                                                 NaN   
2395                                                 NaN   
13370  Pasta, Ravioli, Potato Wedges, Jalapeno Cheese...   
13371  Salads, Veg Buffet, Dahi Vada, Pasta, Vegetari...   
13376                                                NaN   
...                                                  ...   
17397                                                NaN   
17409                                                NaN   
17475                                                NaN   
17479                                                NaN   
23602                                                

## 10.
You and your friend want to go to a Restaurant.<br> 
You would like Pasta and they would like Saag. Get a list of the restaurants that would suit your preferences.<br>
Once you have a list of suitable places select the reataurants that will both take a booking and an online order.

In [111]:
fav = restaurants.menu_item.map(lambda menu: ("Pasta" in menu) & ("Saag" in menu))
result = restaurants[fav].loc[restaurants.book_table == "Yes"].loc[restaurants.online_order=="Yes"]

result

url  \
5564   https://www.zomato.com/bangalore/raaga-indiran...   
10876  https://www.zomato.com/bangalore/brooks-and-bo...   
15121  https://www.zomato.com/bangalore/raaga-indiran...   
17542  https://www.zomato.com/bangalore/srinathjis-in...   
17543  https://www.zomato.com/bangalore/raaga-indiran...   
19884  https://www.zomato.com/bangalore/brooks-and-bo...   
27047  https://www.zomato.com/bangalore/brooks-and-bo...   
28385  https://www.zomato.com/bangalore/the-bier-libr...   
28397  https://www.zomato.com/bangalore/brooks-and-bo...   
29959  https://www.zomato.com/bangalore/brooks-and-bo...   
32624  https://www.zomato.com/bangalore/brooks-and-bo...   
33634  https://www.zomato.com/bangalore/the-bier-libr...   
35270  https://www.zomato.com/bangalore/brooks-and-bo...   
37550  https://www.zomato.com/bangalore/brooks-and-bo...   
38450  https://www.zomato.com/bangalore/the-bangalore...   
38636  https://www.zomato.com/bangalore/the-bangalore...   
39124  https://www.zomato.com/bangalore/eden-park-cun...   
43151  https://www.zomato.com/bangalore/the-bangalore...   
43716  https://www.zomato.com/bangalore/the-bangalore...   
43719  https://www.zomato.com/bangalore/eden-park-cun...   
44180  https://www.zomato.com/bangalore/b1-new-bel-ro...   
45066  https://www.zomato.com/bangalore/srinathjis-in...   
47432  https://www.zomato.com/bangalore/eden-park-cun...   
47792  https://www.zomato.com/bangalore/the-bangalore...   
47982  https://www.zomato.com/bangalore/the-bangalore...   

                                                 address  \
5564   3, Near HDFC Bank, CMH Road, Indiranagar, Bang...   
10876  4, 100 Feet Road, Koramangala 5th Block, Banga...   
15121  3, Near HDFC Bank, CMH Road, Indiranagar, Bang...   
17542  462, Sree Krishna Temple Road, Off 100 Feet Ro...   
17543  3, Near HDFC Bank, CMH Road, Indiranagar, Bang...   
19884  4, 100 Feet Road, Koramangala 5th Block, Banga...   
27047  4, 100 Feet Road, Koramangala 5th Block, Banga...   
28385  14, Patel Narayana Reddy Layout, 80 Ft Main Ro...   
28397  4, 100 Feet Road, Koramangala 5th Block, Banga...   
29959  4, 100 Feet Road, Koramangala 5th Block, Banga...   
32624  4, 100 Feet Road, Koramangala 5th Block, Banga...   
33634  14, Patel Narayana Reddy Layout, 80 Ft Main Ro...   
35270  4, 100 Feet Road, Koramangala 5th Block, Banga...   
37550  4, 100 Feet Road, Koramangala 5th Block, Banga...   
38450   4, KH Road, Double Road, Shanti Nagar, Bangalore   
38636   4, KH Road, Double Road, Shanti Nagar, Bangalore   
39124  21/1, Behind The Hindu, Cunningham Road, Banga...   
43151   4, KH Road, Double Road, Shanti Nagar, Bangalore   
43716   4, KH Road, Double Road, Shanti Nagar, Bangalore   
43719  21/1, Behind The Hindu, Cunningham Road, Banga...   
44180  1259, Opposite Canara Bank, 3rd Block, BEL Lay...   
45066  462, Sree Krishna Temple Road, Off 100 Feet Ro...   
47432  21/1, Behind The Hindu, Cunningham Road, Banga...   
47792   4, KH Road, Double Road, Shanti Nagar, Bangalore   
47982   4, KH Road, Double Road, Shanti Nagar, Bangalore   

                           name online_order book_table    rate  votes  \
5564                      Raaga          Yes        Yes   3.8/5    297   
10876  Brooks and Bonds Brewery          Yes        Yes   4.4/5   3870   
15121                     Raaga          Yes        Yes   3.8/5    298   
17542               Srinathji's          Yes        Yes   3.6/5    592   
17543                     Raaga          Yes        Yes   3.8/5    297   
19884  Brooks and Bonds Brewery          Yes        Yes   4.4/5   3932   
27047  Brooks and Bonds Brewery          Yes        Yes   4.5/5   3944   
28385          The Bier Library          Yes        Yes  4.3 /5    559   
28397  Brooks and Bonds Brewery          Yes        Yes  4.5 /5   3950   
29959  Brooks and Bonds Brewery          Yes        Yes  4.5 /5   3967   
32624  Brooks and Bonds Brewery          Yes        Yes  4.5 /5   3987   
33634          The Bier Library          Ye

In [75]:
restaurants.columns
restaurants['listed_in(city)'].value_counts()

BTM                      3279
Koramangala 7th Block    2938
Koramangala 5th Block    2836
Koramangala 4th Block    2779
Koramangala 6th Block    2623
Jayanagar                2371
JP Nagar                 2096
Indiranagar              1860
Church Street            1827
MG Road                  1811
Brigade Road             1769
Lavelle Road             1744
HSR                      1741
Marathahalli             1659
Residency Road           1620
Whitefield               1620
Bannerghatta Road        1617
Brookefield              1518
Old Airport Road         1425
Kammanahalli             1329
Kalyan Nagar             1309
Basavanagudi             1266
Sarjapur Road            1261
Electronic City          1229
Bellandur                1227
Frazer Town              1185
Malleshwaram             1096
Rajajinagar              1079
Banashankari              863
New BEL Road              740
Name: listed_in(city), dtype: int64